tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_HwenoModifiedSolution.cpp

```diff
@@ -0,0 +1,1290 @@
+    return ScalarTag::type{{{DataVector{1. + 0.1 * x + 0.2 * y + 0.1 * x * y +
+                                        0.1 * x * square(y)}}}};
+  }
+  ();
+
+  const auto lower_xi_vars = [&mesh]() noexcept {
+    Variables<TagsList> result(mesh.number_of_grid_points());
+    // Large values to make sure this neighbor is excluded
+    get(get<ScalarTag>(result)) = 1.e3;
+    return result;
+  }
+  ();
+
+  const auto upper_xi_vars = [&mesh, &logical_coords ]() noexcept {
+    Variables<TagsList> result(mesh.number_of_grid_points());
+    const auto x = get<0>(logical_coords) + 2.;
+    const auto& y = get<1>(logical_coords);
+    get(get<ScalarTag>(result)) =
+        1. + 1. / 3. * x + 0.25 * y - 0.05 * square(x);
+    return result;
+  }
+  ();
+
+  const auto primary_vars = [&mesh, &logical_coords ]() noexcept {
+    Variables<TagsList> result(mesh.number_of_grid_points());
+    const auto& x = get<0>(logical_coords);
+    const auto y = get<1>(logical_coords) - 2.;
+    get(get<ScalarTag>(result)) =
+        1. + 0.25 * x - 0.2 * square(y) + 0.1 * x * square(y);
+    return result;
+  }
+  ();
+
+  const auto upper_eta_vars = [&mesh]() noexcept {
+    Variables<TagsList> result(mesh.number_of_grid_points());
+    // Large values to make sure this neighbor is excluded
+    get(get<ScalarTag>(result)) = 1.e3;
+    return result;
+  }
+  ();
+
+  const auto make_tuple_of_means = [&mesh](
+      const Variables<TagsList>& vars) noexcept {
+    return tuples::TaggedTuple<::Tags::Mean<ScalarTag>>(
+        mean_value(get(get<ScalarTag>(vars)), mesh));
+  };
+
+  struct PackagedData {
+    tuples::TaggedTuple<::Tags::Mean<ScalarTag>> means;
+    Variables<TagsList> volume_data;
+    Mesh<2> mesh;
+  };
+  std::unordered_map<std::pair<Direction<2>, ElementId<2>>, PackagedData,
+                     boost::hash<std::pair<Direction<2>, ElementId<2>>>>
+      neighbor_data{};
+  neighbor_data[lower_xi_neighbor].means = make_tuple_of_means(lower_xi_vars);
+  neighbor_data[lower_xi_neighbor].volume_data = lower_xi_vars;
+  neighbor_data[lower_xi_neighbor].mesh = mesh;
+  neighbor_data[upper_xi_neighbor].means = make_tuple_of_means(upper_xi_vars);
+  neighbor_data[upper_xi_neighbor].volume_data = upper_xi_vars;
+  neighbor_data[upper_xi_neighbor].mesh = mesh;
+  neighbor_data[primary_neighbor].means = make_tuple_of_means(primary_vars);
+  neighbor_data[primary_neighbor].volume_data = primary_vars;
+  neighbor_data[primary_neighbor].mesh = mesh;
+  neighbor_data[upper_eta_neighbor].means = make_tuple_of_means(upper_eta_vars);
+  neighbor_data[upper_eta_neighbor].volume_data = upper_eta_vars;
+  neighbor_data[upper_eta_neighbor].mesh = mesh;
+
+  ScalarTag::type modified_tensor;
+  SlopeLimiters::compute_hweno_modified_neighbor_solution<ScalarTag>(
+      make_not_null(&modified_tensor), local_tensor, element, mesh,
+      neighbor_data, primary_neighbor);
+
+  // The expected coefficient values for the result of the constrained fit are
```

Comment:
I think both the C++ and the mathematica could be cleaner using a vector of values, and an accumulation function which adds `v[i] * pow(x, i%number_of_x) * pow(y, i/number_of_y)`.
However, I don't think it will be much cleaner, there is an intrinsic messiness, and may be more effort than it is worth, so I'll leave this up to your discretion.

Reply:
I don't disagree, but I am also not particularly motivated to make this change 😛

Reply:
In case you feel motivated to copy-paste C++ code :) (I mostly wanted to make sure my regex replace skills don't get rusty :P )
```cpp
constexpr std::array<double, 12> c{{
    757538269. / 712675275., 92367. / 360620.,     -1659. / 558961.,
    -6083. / 558961.,        1135772. / 18273725., 42. / 6935.,
    -126. / 42997.,          -462. / 42997.,       -44104369. / 237558425.,
    18283. / 180310.,        -378. / 558961.,      -1386. / 558961.}};
return Scalar<DataVector>{{{DataVector{
    c[0] + c[1] * x + c[2] * square(x) + c[3] * cube(x) +
    y * (c[4] + c[5] * x + c[6] * square(x) + c[7] * cube(x)) +
    square(y) * (c[8] + c[9] * x + c[10] * square(x) + c[11] * cube(x))}}}};
```